python/pyspark/sql/pandas/types.py

```diff
@@ -221,6 +221,15 @@ def to_arrow_schema(
     return pa.schema(fields)
 
 
+def is_variant(at: "pa.DataType") -> bool:
+    """Check if a PyArrow struct data type represents a variant"""
+    import pyarrow.types as types
+    assert types.is_struct(at)
```

Comment:
Why is this an assert? Should this just return false if it is not a struct?

Reply:
This function is intended to be called with `at.is_struct` so I want to prevent developers from using this function with non-struct types. I should add a comment.

Checking `is_struct` in this function adds cost in production (where I'm assuming Python runs in optimized mode so asserts are disabled).